pkg/ccl/streamingccl/streamclient/client.go

```diff
@@ -161,6 +161,9 @@ func NewStreamClient(
 	case "postgres", "postgresql":
 		// The canonical PostgreSQL URL scheme is "postgresql", however our
 		// own client commands also accept "postgres".
+		if processOptions(opts).forSpanConfigs {
+			return NewSpanConfigStreamClient(ctx, streamURL, opts...)
+		}
```

Comment:
[nit] I would mildly prefer that the caller just call NewSpanConfigStreamClient directly. That sets us up in the right place for it being its own type in the future.

If we leave it as is, I think we need to at least update the `external` branch of this switch statement to make sure it it passing the opts through.